apps/src/music/FieldSounds.js

```diff
@@ -0,0 +1,112 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import SoundsPanel from './SoundsPanel';
+import GoogleBlockly from 'blockly/core';
+
+class FieldSounds extends GoogleBlockly.Field {
+  constructor(options) {
+    super(options.currentValue);
+
+    this.options = options;
+    this.playingPreview = null;
+    this.SERIALIZABLE = true;
+    this.CURSOR = 'default';
+  }
+
+  saveState() {
+    return this.getValue();
+  }
+
+  loadState(state) {
+    this.setValue(state);
+  }
+
+  static fromJson(options) {
+    return new FieldSounds(options);
+  }
+
+  showEditor_() {
+    super.showEditor_();
+
+    const editor = this.dropdownCreate_();
+    Blockly.DropDownDiv.getContentDiv().appendChild(editor);
+
+    Blockly.DropDownDiv.setColour(
+      this.sourceBlock_.style.colourPrimary,
+      this.sourceBlock_.style.colourTertiary
+    );
+
+    Blockly.DropDownDiv.showPositionedByField(
+      this,
+      this.dropdownDispose_.bind(this)
+    );
+  }
+
+  dropdownCreate_() {
+    this.newDiv_ = document.createElement('div');
+
+    this.renderContent();
+
+    this.newDiv_.style.color = 'white';
```

Comment:
To me, this color makes the field look it should be a text input field. Blockly does have a `block.getColour()` method that can be used to make fields match the color of the block.

Reply:
I took inspiration from the [dropdown](https://github.com/google/blockly/blob/1f862cb878f7eec36b71c638b85d5199bff01fcb/core/field_dropdown.ts) implementation to get the field styled in the colour of the block.  I looked at also drawing the dropdown arrow, which would probably help, but would need a _lot_ more code if we use the same implementation, so I'm passing on that for now.